### ISSUE TITLE
Problem with failed transactions

### ISSUE THREAD

**Issue by hommyshop** [wirecard/shopify-ee#7]
If a transaction gets declined / customer fails to pay, they are simply redirected to the Shopify's Thank you page and a new successful order gets created on our Shopify store. There are no errors on the Shopify's side. The only way to notice is by going to WEP dashboard and match the transaction ID. (Please notice a related issue, where transaction IDs are missing, making this task even more difficult).

How should we scale it to thousands of transactions? Should we manually check every transaction to make sure they are paid? And should we also contact customers whose payments have failed and ask them to try again? Meanwhile the customer gets an email about a successful purchase, when in reality they were unable to pay for it. Is there a better way to go about this problem?

I have also noticed a direct integration option on Shopify. Is it working? Can we get credentials for this one? Is it better integrated and will not have such issues?

![screenshot-2019-04-13:13:20:53](https://user-images.githubusercontent.com/49198848/56078330-6ab00c00-5def-11e9-9282-54b8c7f989e6.png)

**Comment by jakubpolomsky** [comment]
Please provide me some steps to reproduce. After a failed payment we always get to the payment selection page and no new order gets created. I will have a look at it immediately after I know how to. To the direct card integration - I have forwarded the question to our support department (as on github we handle only technical questions) and the colleagues will contact you.

Kind regards and thanks for reporting!

**Comment by hommyshop** [comment]
It is difficult to test the checkout as no test credentials seem to work with this plugin for some reason. So I am testing with live traffic and my own card.

Currently I am trying to replicate the problem and noticed another problem. Every time I click the blue "Payment" button in the checkout, and if the card gets immediately declined, the page does not redirect and does not display any error. In the network requests I can see that the form gets submitted, and on WEP the payment activity is immediately displayed correctly as Failed.

![image](https://user-images.githubusercontent.com/49198848/56134637-ae457a00-5f97-11e9-93cd-02c068c00e44.png)

![image](https://user-images.githubusercontent.com/49198848/56134761-e6e55380-5f97-11e9-9b8e-4117048d2244.png)

However, if the card is accepted then the payment goes through and gets redirected to Shopify's processing page, and then to the Thank you page as expected.

I believe the problem with orders getting created incorrectly happens later, if the payment is not immediately accepted. The first time I noticed it is when I tried to checkout with a bigger amount and triggered the 3D page to enter my password. Then I cancelled the 3D page and instead of an error, I got to the thank you page with the order getting created on Shopify as paid.

![image](https://user-images.githubusercontent.com/49198848/56135532-a090f400-5f99-11e9-9f19-e23ecfead91a.png)
This one was cancelled at the 3D step, shows failed in WEP "500.1076 - Customer failed or Cancelled authentication", however was accepted as Paid in Shopify.

The second time it happened, when a real customer tried checking out, their card was initially accepted, they got to the thank you page, with their order being created on Shopify. However, in WEP their card was rejected due to Insufficient balance. 

![image](https://user-images.githubusercontent.com/49198848/56135320-21032500-5f99-11e9-9743-771d656918ca.png)
This order got accepted in Shopify, and rejected in WEP as 500.2401 - insufficient funds/ no funds
